### ISSUE TITLE
Add flag to include DNSSEC additional records in output

### ISSUE THREAD

**Issue by secastro** [zmap/zdns#318]
More than an issue, this is a feature request, if it doesn't exist.

Is it possible to add a flag that will save, as an additional record, DNSSEC-related records when the destination nameserver is doing validation?

We run a large-scale collection for a ccTLD, and saving the responses and the RRSIG records could allow us to validate those signatures offline.

**Comment by phillip-stephens** [comment]
Looks like this was resolved by #323 

When I compare `dig +dnssec` to `./zdns --dnssec` I'm getting the same response when querying for `cloudflare.com`

### dig
```
;; QUESTION SECTION:
;cloudflare.com.                        IN      A

;; ANSWER SECTION:
cloudflare.com.         300     IN      A       104.16.132.229
cloudflare.com.         300     IN      A       104.16.133.229
cloudflare.com.         300     IN      RRSIG   A 13 2 300 20240718162838 20240716142838 34505 cloudflare.com. 0bjO17rcDVjSNVLZGRCzc692BQeAHkP2pirKSPVMbnmec1ryo2AFQkyb znaXlpYxestJTpxzeqmdjLQUMWU1gA==
```

### zdns
```
$ echo "cloudflare.com" | ./zdns A --dnssec | jq
{
  "data": {
    "additionals": [
      {
        "flags": "do",
        "type": "EDNS0",
        "udpsize": 1232,
        "version": 0
      }
    ],
    "answers": [
      {
        "answer": "104.16.133.229",
        "class": "IN",
        "name": "cloudflare.com",
        "ttl": 125,
        "type": "A"
      },
      {
        "answer": "104.16.132.229",
        "class": "IN",
        "name": "cloudflare.com",
        "ttl": 125,
        "type": "A"
      },
      {
        "algorithm": 13,
        "class": "IN",
        "expiration": "20240718162846",
        "inception": "20240716142846",
        "keytag": 34505,
        "labels": 2,
        "name": "cloudflare.com",
        "original_ttl": 300,
        "signature": "GhQaXcxq+BmqR8ap8H4E604+80a0KaOSfLlYABo9rCC1OdFQhZXWQznntsccYgTn83VF/ct1HEpEcxcB6NpChg==",
        "signer_name": "cloudflare.com.",
        "ttl": 125,
        "type": "RRSIG",
        "type_covered": 1
      }
    ],
    "protocol": "udp",
    "resolver": "1.1.1.1:53"
  },
  "name": "cloudflare.com",
  "status": "NOERROR",
  "timestamp": "2024-07-17T11:32:13-04:00"
}
```

@zakird We don't need to try to pull the `DNSKEY` records or anything for this, do we?

**Comment by zakird** [comment]
I agree this looks done. It would be nice to one day be able to validate DNSSEC responses, but that seems beyond the scope of this particular issue.
